File: 02_Stacks/02_Evaluation_of_Postfix/main.py
# Main Function
def evaluatePostfix(exp):
    # Create stack representation
    stack = []

    # Iterate over the expression to scan the characters
    for ch in exp:
        # If the scanned character is Number
        if ch.isdigit():
            stack.append(ch)
        
        # If the scanned character is operator
        else:
            val1 = stack.pop(-1)
            val2 = stack.pop(-1)
            stack.append(str(eval(val2 + ch + val1)))
        
    return int(stack.pop())

# Driver Program
exp = "231*+9-"
print("Postfix evaluation: {}".format(evaluatePostfix(exp)))

File: 02_Stacks/02_Evaluation_of_Postfix/main.cpp
#include<iostream>
#include<string.h>
#include<stack>

// Main Funtion
int evaluatePostfix(std::string exp){
    // Create the stack
    std::stack<int> stack;

    // Scan the characters
    for(char &ch: exp){
        // If the scanned character is a Number
        if(isdigit(ch))
            stack.push(ch - '0');
        
        // If the scanned character is an operator
        else{
            int val1 = stack.top(); stack.pop();
            int val2 = stack.top(); stack.pop();
            switch(ch){
            case '+': stack.push(val2 + val1); break;  
            case '-': stack.push(val2 - val1); break;  
            case '*': stack.push(val2 * val1); break;  
            case '/': stack.push(val2 / val1); break;  
            }
        }
    }

    return stack.top();
}

// Driver Program
int main()  {  
    std::string exp = "231*+9-";  
    std::cout<<"postfix evaluation: "<< evaluatePostfix(exp);  
    std::cout<<std::endl;
    return 0;  
}  